### PR TITLE
Improving performance by reducing memory allocations and task executions

### PR DESCRIPTION
**Pull request by supunkamburugamuve**

### CODE REVIEW

File: twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java [hunk 1]
```diff
@@ -292,16 +298,31 @@ public boolean writeEnd(String edge, Object key, Object message) {
...
+
+    checkAllEdgesFinished();
     return collect;
   }
 
+  private void checkAllEdgesFinished() {
+    boolean finished = true;
+    for (String e : outEdgeNames) {
+      if (!isDone.containsKey(e)) {
+        finished = false;
```
Comment (chathurawidanage) [inline twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java]: Could break here?

File: twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java [hunk 2]
```diff
@@ -292,16 +298,31 @@ public boolean writeEnd(String edge, Object key, Object message) {
...
+      }
+    }
+    if (finished) {
```
Comment (chathurawidanage) [inline twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java]: Can get rid of if block?

Comment (supunkamburugamuve) [reply twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java]: These changes are in the code

File: twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java [hunk 3]
```diff
@@ -188,7 +194,7 @@ public int getParallelism() {
     return parallelism;
   }
 
-  /**
+  /**Bloomington
```
Comment (chathurawidanage) [inline twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java]: :-D

Comment (supunkamburugamuve) [reply twister2/executor/src/java/edu/iu/dsc/tws/executor/core/TaskContextImpl.java]: Removed this one :)

**Pull request merged**